app/controllers/application_pets_controller.rb

```diff
@@ -0,0 +1,27 @@
+class ApplicationPetsController < ApplicationController
+  
+  def new
+  end
+
+  def create
+    @application_pet = ApplicationPet.new(application_pets_params)
+    @application_pet.save
+    redirect_to "/applications/#{@application_pet.application_id}"
+  end
+
+  def update
+    if params[:status] == "approve"
+      application_pet = ApplicationPet.find(params[:id])
+      application_pet.update(status: "Approved")
+    elsif params[:status] == "reject"
+      application_pet = ApplicationPet.find(params[:id])
+      application_pet.update(status: "Rejected")
+    end
+    redirect_to "/admin/applications/#{params[:application_id]}"
+  end
```

Comment:
This method is excellent, my only suggestion would be to eliminate the repetition you have on lines 14 and 17 since in both branches you need to find the ApplicationPet record so you can do that outside of the if.

```ruby
def update
  application_pet = ApplicationPet.find(params[:id])

  if params[:status] == "approve"
    application_pet.update(status: "Approved")
  elsif params[:status] == "reject"
    application_pet.update(status: "Rejected")
  end

  redirect_to "/admin/applications/#{params[:application_id]}"
end
```